homeassistant/components/light/ihc.py

```diff
@@ -0,0 +1,123 @@
+                vol.Optional(CONF_DIMMABLE, default=False): cv.boolean,
+            }, validate_name)
+        ])
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the ihc lights platform."""
+    ihc_controller = hass.data[IHC_DATA][IHC_CONTROLLER]
+    devices = []
+    if discovery_info:
+        for name, device in discovery_info.items():
+            ihc_id = device['ihc_id']
+            product_cfg = device['product_cfg']
+            product = device['product']
+            light = IhcLight(ihc_controller, name, ihc_id,
+                             product_cfg[CONF_DIMMABLE], product)
+            devices.append(light)
+    else:
+        lights = config[CONF_LIGHTS]
+        for light in lights:
+            ihc_id = light[CONF_ID]
+            name = light[CONF_NAME]
+            dimmable = light[CONF_DIMMABLE]
+            device = IhcLight(ihc_controller, name, ihc_id, dimmable)
+            devices.append(device)
+
+    add_devices(devices)
+
+
+class IhcLight(IHCDevice, Light):
+    """Representation of a IHC light.
+
+    For dimmable lights, the associated IHC resource should be a light
+    level (integer). For non dimmable light the IHC resource should be
+    an on/off (boolean) resource
+    """
+
+    def __init__(self, ihc_controller, name, ihc_id, dimmable=False,
+                 product: Element=None):
+        """Initialize the light."""
+        super().__init__(ihc_controller, name, ihc_id, product)
+        self._brightness = 0
+        self._dimmable = dimmable
+        self._state = None
+
+    @property
+    def brightness(self) -> int:
+        """Return the brightness of this light between 0..255."""
+        return self._brightness
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if light is on."""
+        return self._state
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        if self._dimmable:
+            return SUPPORT_BRIGHTNESS
+        return 0
+
+    def turn_on(self, **kwargs) -> None:
+        """Turn the light on."""
+        if ATTR_BRIGHTNESS in kwargs:
+            brightness = kwargs[ATTR_BRIGHTNESS]
+        else:
+            brightness = self._brightness
+            if brightness == 0:
+                brightness = 255
+
+        if self._dimmable:
+            self.ihc_controller.set_runtime_value_int(
+                self.ihc_id, int(brightness * 100 / 255))
+        else:
+            self.ihc_controller.set_runtime_value_bool(self.ihc_id, True)
+
+    def turn_off(self, **kwargs) -> None:
+        """Turn the light off."""
+        if self._dimmable:
+            self.ihc_controller.set_runtime_value_int(self.ihc_id, 0)
+        else:
+            self.ihc_controller.set_runtime_value_bool(self.ihc_id, False)
+
+    def on_ihc_change(self, ihc_id, value):
+        """Callback from Ihc notifications."""
+        if type(value) is int:
```

Comment:
```py
if isinstance(value, int):
```

Reply:
```python
if isinstance(value, int):
```
is not working because if value is bool it will also return true. I have reversed to 

```python
if isinstance(value, bool):
```

Then it is working because - when value is int it will return false. (Now I can remove the #pylint disable)